client-spark/common/src/main/java/org/apache/spark/shuffle/RssSparkConfig.java

```diff
@@ -37,6 +37,28 @@
 
 public class RssSparkConfig {
 
+  public static final ConfigOption<Boolean> RSS_RESUBMIT_STAGE_ENABLED =
+      ConfigOptions.key("rss.stageRetry.enabled")
+          .booleanType()
+          .defaultValue(false)
+          .withDeprecatedKeys(RssClientConfig.RSS_RESUBMIT_STAGE)
+          .withDescription("Whether to enable the resubmit stage for fetch/write failure");
+
+  public static final ConfigOption<Boolean> RSS_RESUBMIT_STAGE_WITH_FETCH_FAILURE_ENABLED =
+      ConfigOptions.key("rss.stageRetry.fetchFailureEnabled")
+          .booleanType()
+          .defaultValue(false)
+          .withFallbackKeys(RSS_RESUBMIT_STAGE_ENABLED.key(), RssClientConfig.RSS_RESUBMIT_STAGE)
+          .withDescription(
+              "Activate the stage retry when fetching failure occurs when it's enabled.");
```

Comment:
If set to true, the stage retry mechanism will be enabled when a fetch failure occurs.